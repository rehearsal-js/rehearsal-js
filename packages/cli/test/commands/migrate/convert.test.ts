import { resolve } from 'path';
import { readdirSync, readJSONSync } from 'fs-extra';
import { setGracefulCleanup } from 'tmp';
import { beforeEach, describe, expect, test } from 'vitest';
import { type SimpleGit, simpleGit, type SimpleGitOptions } from 'simple-git';

import { runBin, prepareTmpDir } from '../../test-helpers';

setGracefulCleanup();

// These tests are using `runBin` to call the entire `migrate` command instead of testing the individual convert task
// Since there are a lot of context needs to be injected
// And it does make sense to run the whole cli for this case
describe('Task: convert', async () => {
  let basePath = '';

  beforeEach(() => {
    basePath = prepareTmpDir('basic');
  });

  test('able to migrate from default all files .js in root', async () => {
    const result = await runBin('migrate', [], {
      cwd: basePath,
    });

    // Test summary message
    expect(result.stdout).toContain(`3 JS files converted to TS`);

    const fileList = readdirSync(basePath);

    expect(fileList).toContain('index.ts');
    expect(fileList).toContain('foo.ts');
    expect(fileList).toContain('depends-on-foo.ts');
    expect(fileList).not.toContain('index.js');
    expect(fileList).not.toContain('foo.js');
    expect(fileList).not.toContain('depends-on-foo.js');

    const config = readJSONSync(resolve(basePath, 'tsconfig.json'));
    expect(config.include).toContain('index.ts');
    expect(config.include).toContain('foo.ts');
    expect(config.include).toContain('depends-on-foo.ts');
  });

  test('able to migrate from specific entrypoint', async () => {
    const result = await runBin('migrate', ['--entrypoint', 'depends-on-foo.js'], {
      cwd: basePath,
    });

    expect(result.stdout).toContain(`2 JS files converted to TS`);

    const fileList = readdirSync(basePath);
    expect(fileList).toContain('depends-on-foo.ts');
    expect(fileList).toContain('foo.ts');

    expect(fileList).not.toContain('depends-on-foo.js');
    expect(fileList).not.toContain('foo.js');

    const config = readJSONSync(resolve(basePath, 'tsconfig.json'));
    expect(config.include).toContain('depends-on-foo.ts');
    expect(config.include).toContain('foo.ts');
  });

  test('Print debug messages with verbose', async () => {
    const result = await runBin('migrate', ['--verbose'], {
      cwd: basePath,
    });

    expect(result.stdout).toContain(`\x1B[34mdebug\x1B[39m`);
  });

  test('Generate sarif report by default', async () => {
    await runBin('migrate', [], {
      cwd: basePath,
    });

    const reportPath = resolve(basePath, '.rehearsal');

    expect(readdirSync(reportPath)).toContain('migrate-report.sarif');
  });

  test('stage report if in a git repo', async () => {
    // simulate clean git project
    const git: SimpleGit = simpleGit({
      baseDir: basePath,
    } as Partial<SimpleGitOptions>);
    // Init git, add and commit existed files, to make it a clean state
    await git.init();
    await git.add(readdirSync(basePath));
    // GH CI would require git name and email
    await git.addConfig('user.name', 'tester');
    await git.addConfig('user.email', 'tester@tester.com');
    await git.commit('foo');

    await runBin('migrate', [], {
      cwd: basePath,
    });

    const reportPath = resolve(basePath, '.rehearsal');
    expect(readdirSync(reportPath)).toContain('migrate-report.sarif');

    const gitStatus = await git.status();
    expect(gitStatus.staged).toStrictEqual([
      '.eslintrc.js',
      '.rehearsal-eslintrc.js',
      '.rehearsal/migrate-report.sarif',
      'tsconfig.json',
    ]);
  });

  test('Generate report in different formats with -f flag', async () => {
    await runBin('migrate', ['-f', 'json,md,sarif,foo'], {
      cwd: basePath,
    });

    const reportPath = resolve(basePath, '.rehearsal');

    const reportList = readdirSync(reportPath);

    expect(reportList).toContain('migrate-report.json');
    expect(reportList).toContain('migrate-report.md');
    expect(reportList).toContain('migrate-report.sarif');
    expect(reportList).not.toContain('migrate-report.foo');
  });

  test('Run cli againt specific basePath via -basePath option', async () => {
    basePath = prepareTmpDir('custom_basepath');

    const customBasePath = resolve(basePath, 'base');
    const result = await runBin('migrate', ['--basePath', customBasePath]);

    const fileList = readdirSync(customBasePath);

    expect(result.stdout).toContain('Install dependencies');
    expect(result.stdout).toContain('Create tsconfig.json');
    expect(fileList).toContain('tsconfig.json');

    expect(result.stdout).toContain(`1 JS file converted to TS`);
    expect(fileList).toContain('index.ts');
    expect(fileList).not.toContain('index.js');

    const config = readJSONSync(resolve(customBasePath, 'tsconfig.json'));
    expect(config.include).toContain('index.ts');
  });
});
