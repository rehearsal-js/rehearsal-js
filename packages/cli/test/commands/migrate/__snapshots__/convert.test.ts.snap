// Vitest Snapshot v1

exports[`Task: convert > accept and discard changes with git 1`] = `
"[STARTED] Initialize
[DATA] ? We have found multiple packages in your project, select the one to migrate:  
[DATA]  basic(no progress found)
[DATA]  We have found multiple packages in your project, select the one to migrate:  basic(no progress found)
[DATA] Running migration on <tmp-path>
[SUCCESS] Initialize
[STARTED] Install dependencies
[DATA] Installing devDependency @types/node...
[DATA] Installing devDependency @typescript-eslint/eslint-plugin...
[DATA] Installing devDependency @typescript-eslint/parser...
[DATA] Installing devDependency eslint...
[DATA] Installing devDependency eslint-config-prettier...
[DATA] Installing devDependency eslint-plugin-prettier...
[DATA] Installing devDependency prettier...
[DATA] Installing devDependency typescript...
[SUCCESS] Install dependencies
[STARTED] Create tsconfig.json
[SUCCESS] Create tsconfig.json
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
[STARTED] Convert JS files to TS
[DATA] git mv /foo.js to /foo.ts
[DATA] processing file: /foo.ts
[DATA] ? Please view the migration changes for <tmp-path>/foo.js and select an option to continue:
[DATA]  Accept
[DATA]   Edit
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/foo.js and select an option to continue:
[DATA]   Accept
[TITLE] Migration Complete
[TITLE]   1 JS file converted to TS
[TITLE]   1 errors caught by rehearsal
[TITLE]   1 have been fixed by rehearsal
[TITLE]   0 errors need to be fixed manually
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE]     -- 0 eslint errors, with details in the report
[DATA] git mv /depends-on-foo.js to /depends-on-foo.ts
[DATA] processing file: /foo.ts
[DATA] processing file: /depends-on-foo.ts
[DATA] ? Please view the migration changes for <tmp-path>/depends-on-foo.js and select an option to continue:
[DATA]  Accept
[DATA]   Edit
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/depends-on-foo.js and select an option to continue:
[DATA]   Accept
[TITLE] Migration Complete
[TITLE]   2 JS files converted to TS
[TITLE]   2 errors caught by rehearsal
[TITLE]   1 have been fixed by rehearsal
[TITLE]   1 errors need to be fixed manually
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE]     -- 1 eslint errors, with details in the report
[DATA] git mv /index.js to /index.ts
[DATA] processing file: /foo.ts
[DATA] processing file: /depends-on-foo.ts
[DATA] processing file: /index.ts
[DATA] ? Please view the migration changes for <tmp-path>/index.js and select an option to continue:
[DATA]  Accept
[DATA]   Edit
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/index.js and select an option to continue:
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/index.js and select an option to continue:
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/index.js and select an option to continue:
[DATA]   Discard
[TITLE] Migration Complete
[TITLE]   3 JS files converted to TS
[TITLE]   4 errors caught by rehearsal
[TITLE]   3 have been fixed by rehearsal
[TITLE]   1 errors need to be fixed manually
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE]     -- 1 eslint errors, with details in the report
[SUCCESS] Migration Complete
[SUCCESS]   3 JS files converted to TS
[SUCCESS]   4 errors caught by rehearsal
[SUCCESS]   3 have been fixed by rehearsal
[SUCCESS]   1 errors need to be fixed manually
[SUCCESS]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[SUCCESS]     -- 1 eslint errors, with details in the report"
`;

exports[`Task: convert > accept changes without git 1`] = `
"[STARTED] Initialize
[DATA] ? We have found multiple packages in your project, select the one to migrate:  
[DATA]  basic(no progress found)
[DATA]  We have found multiple packages in your project, select the one to migrate:  basic(no progress found)
[DATA] Running migration on <tmp-path>
[SUCCESS] Initialize
[STARTED] Install dependencies
[DATA] Installing devDependency @types/node...
[DATA] Installing devDependency @typescript-eslint/eslint-plugin...
[DATA] Installing devDependency @typescript-eslint/parser...
[DATA] Installing devDependency eslint...
[DATA] Installing devDependency eslint-config-prettier...
[DATA] Installing devDependency eslint-plugin-prettier...
[DATA] Installing devDependency prettier...
[DATA] Installing devDependency typescript...
[SUCCESS] Install dependencies
[STARTED] Create tsconfig.json
[SUCCESS] Create tsconfig.json
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
[STARTED] Convert JS files to TS
[DATA] git mv /foo.js to /foo.ts
[DATA] processing file: /foo.ts
[DATA] ? Please view the migration changes for <tmp-path>/foo.js and select an option to continue:
[DATA]  Accept
[DATA]   Edit
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/foo.js and select an option to continue:
[DATA]   Accept
[TITLE] Migration Complete
[TITLE]   1 JS file converted to TS
[TITLE]   1 errors caught by rehearsal
[TITLE]   1 have been fixed by rehearsal
[TITLE]   0 errors need to be fixed manually
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE]     -- 0 eslint errors, with details in the report
[DATA] git mv /depends-on-foo.js to /depends-on-foo.ts
[DATA] processing file: /foo.ts
[DATA] processing file: /depends-on-foo.ts
[DATA] ? Please view the migration changes for <tmp-path>/depends-on-foo.js and select an option to continue:
[DATA]  Accept
[DATA]   Edit
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/depends-on-foo.js and select an option to continue:
[DATA]   Accept
[TITLE] Migration Complete
[TITLE]   2 JS files converted to TS
[TITLE]   2 errors caught by rehearsal
[TITLE]   1 have been fixed by rehearsal
[TITLE]   1 errors need to be fixed manually
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE]     -- 1 eslint errors, with details in the report
[DATA] git mv /index.js to /index.ts
[DATA] processing file: /foo.ts
[DATA] processing file: /depends-on-foo.ts
[DATA] processing file: /index.ts
[DATA] ? Please view the migration changes for <tmp-path>/index.js and select an option to continue:
[DATA]  Accept
[DATA]   Edit
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/index.js and select an option to continue:
[DATA]   Accept
[TITLE] Migration Complete
[TITLE]   3 JS files converted to TS
[TITLE]   4 errors caught by rehearsal
[TITLE]   3 have been fixed by rehearsal
[TITLE]   1 errors need to be fixed manually
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE]     -- 1 eslint errors, with details in the report
[SUCCESS] Migration Complete
[SUCCESS]   3 JS files converted to TS
[SUCCESS]   4 errors caught by rehearsal
[SUCCESS]   3 have been fixed by rehearsal
[SUCCESS]   1 errors need to be fixed manually
[SUCCESS]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[SUCCESS]     -- 1 eslint errors, with details in the report"
`;

exports[`Task: convert > cancel prompt in interactive mode 1`] = `
"[STARTED] Initialize
[DATA] ? We have found multiple packages in your project, select the one to migrate:  
[DATA]  basic(no progress found)
[DATA]  We have found multiple packages in your project, select the one to migrate:  basic(no progress found)
[DATA] Running migration on <tmp-path>
[SUCCESS] Initialize
[STARTED] Install dependencies
[DATA] Installing devDependency @types/node...
[DATA] Installing devDependency @typescript-eslint/eslint-plugin...
[DATA] Installing devDependency @typescript-eslint/parser...
[DATA] Installing devDependency eslint...
[DATA] Installing devDependency eslint-config-prettier...
[DATA] Installing devDependency eslint-plugin-prettier...
[DATA] Installing devDependency prettier...
[DATA] Installing devDependency typescript...
[SUCCESS] Install dependencies
[STARTED] Create tsconfig.json
[SUCCESS] Create tsconfig.json
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
[STARTED] Convert JS files to TS
[DATA] git mv /foo.js to /foo.ts
[DATA] processing file: /foo.ts
[DATA] ? Please view the migration changes for <tmp-path>/foo.js and select an option to continue:
[DATA]  Accept
[DATA]   Edit
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/foo.js and select an option to continue:
[FAILED] Cancelled prompt."
`;

exports[`Task: convert > generate reports 1`] = `
"[STARTED] Initialize
[DATA] Running migration on basic
[SUCCESS] Initialize
[STARTED] Install dependencies
[DATA] Installing devDependency @types/node...
[DATA] Installing devDependency @typescript-eslint/eslint-plugin...
[DATA] Installing devDependency @typescript-eslint/parser...
[DATA] Installing devDependency eslint...
[DATA] Installing devDependency eslint-config-prettier...
[DATA] Installing devDependency eslint-plugin-prettier...
[DATA] Installing devDependency prettier...
[DATA] Installing devDependency typescript...
[SUCCESS] Install dependencies
[STARTED] Create tsconfig.json
[SUCCESS] Create tsconfig.json
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
[STARTED] Convert JS files to TS
[DATA] git mv /foo.js to /foo.ts
[DATA] git mv /depends-on-foo.js to /depends-on-foo.ts
[DATA] git mv /index.js to /index.ts
[DATA] processing file: /foo.ts
[DATA] processing file: /depends-on-foo.ts
[DATA] processing file: /index.ts
[TITLE] Migration Complete
[TITLE] 
[TITLE] 
[TITLE]   3 JS files converted to TS
[TITLE] 
[TITLE]   4 errors caught by rehearsal
[TITLE] 
[TITLE]   3 have been fixed by rehearsal
[TITLE] 
[TITLE]   1 errors need to be fixed manually
[TITLE] 
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE] 
[TITLE]     -- 1 eslint errors, with details in the report
[TITLE] 
[SUCCESS] Migration Complete
[SUCCESS] 
[SUCCESS] 
[SUCCESS]   3 JS files converted to TS
[SUCCESS] 
[SUCCESS]   4 errors caught by rehearsal
[SUCCESS] 
[SUCCESS]   3 have been fixed by rehearsal
[SUCCESS] 
[SUCCESS]   1 errors need to be fixed manually
[SUCCESS] 
[SUCCESS]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[SUCCESS] 
[SUCCESS]     -- 1 eslint errors, with details in the report
[SUCCESS] 
"
`;

exports[`Task: convert > migrate from default all files .js in root 1`] = `
"[STARTED] Initialize
[DATA] Running migration on basic
[SUCCESS] Initialize
[STARTED] Install dependencies
[DATA] Installing devDependency @types/node...
[DATA] Installing devDependency @typescript-eslint/eslint-plugin...
[DATA] Installing devDependency @typescript-eslint/parser...
[DATA] Installing devDependency eslint...
[DATA] Installing devDependency eslint-config-prettier...
[DATA] Installing devDependency eslint-plugin-prettier...
[DATA] Installing devDependency prettier...
[DATA] Installing devDependency typescript...
[SUCCESS] Install dependencies
[STARTED] Create tsconfig.json
[SUCCESS] Create tsconfig.json
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
[STARTED] Convert JS files to TS
[DATA] git mv /foo.js to /foo.ts
[DATA] git mv /depends-on-foo.js to /depends-on-foo.ts
[DATA] git mv /index.js to /index.ts
[DATA] processing file: /foo.ts
[DATA] processing file: /depends-on-foo.ts
[DATA] processing file: /index.ts
[TITLE] Migration Complete
[TITLE] 
[TITLE] 
[TITLE]   3 JS files converted to TS
[TITLE] 
[TITLE]   4 errors caught by rehearsal
[TITLE] 
[TITLE]   3 have been fixed by rehearsal
[TITLE] 
[TITLE]   1 errors need to be fixed manually
[TITLE] 
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE] 
[TITLE]     -- 1 eslint errors, with details in the report
[TITLE] 
[SUCCESS] Migration Complete
[SUCCESS] 
[SUCCESS] 
[SUCCESS]   3 JS files converted to TS
[SUCCESS] 
[SUCCESS]   4 errors caught by rehearsal
[SUCCESS] 
[SUCCESS]   3 have been fixed by rehearsal
[SUCCESS] 
[SUCCESS]   1 errors need to be fixed manually
[SUCCESS] 
[SUCCESS]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[SUCCESS] 
[SUCCESS]     -- 1 eslint errors, with details in the report
[SUCCESS] 
"
`;

exports[`Task: convert > migrate from specific entrypoint 1`] = `
"[STARTED] Initialize
[DATA] Running migration on basic
[SUCCESS] Initialize
[STARTED] Install dependencies
[DATA] Installing devDependency @types/node...
[DATA] Installing devDependency @typescript-eslint/eslint-plugin...
[DATA] Installing devDependency @typescript-eslint/parser...
[DATA] Installing devDependency eslint...
[DATA] Installing devDependency eslint-config-prettier...
[DATA] Installing devDependency eslint-plugin-prettier...
[DATA] Installing devDependency prettier...
[DATA] Installing devDependency typescript...
[SUCCESS] Install dependencies
[STARTED] Create tsconfig.json
[SUCCESS] Create tsconfig.json
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
[STARTED] Convert JS files to TS
[DATA] git mv /foo.js to /foo.ts
[DATA] git mv /depends-on-foo.js to /depends-on-foo.ts
[DATA] processing file: /foo.ts
[DATA] processing file: /depends-on-foo.ts
[TITLE] Migration Complete
[TITLE] 
[TITLE] 
[TITLE]   2 JS files converted to TS
[TITLE] 
[TITLE]   2 errors caught by rehearsal
[TITLE] 
[TITLE]   1 have been fixed by rehearsal
[TITLE] 
[TITLE]   1 errors need to be fixed manually
[TITLE] 
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE] 
[TITLE]     -- 1 eslint errors, with details in the report
[TITLE] 
[SUCCESS] Migration Complete
[SUCCESS] 
[SUCCESS] 
[SUCCESS]   2 JS files converted to TS
[SUCCESS] 
[SUCCESS]   2 errors caught by rehearsal
[SUCCESS] 
[SUCCESS]   1 have been fixed by rehearsal
[SUCCESS] 
[SUCCESS]   1 errors need to be fixed manually
[SUCCESS] 
[SUCCESS]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[SUCCESS] 
[SUCCESS]     -- 1 eslint errors, with details in the report
[SUCCESS] 
"
`;

exports[`Task: convert > view changes in $EDITOR 1`] = `
"[STARTED] Initialize
[DATA] ? We have found multiple packages in your project, select the one to migrate:  
[DATA]  basic(no progress found)
[DATA]  We have found multiple packages in your project, select the one to migrate:  basic(no progress found)
[DATA] Running migration on <tmp-path>
[SUCCESS] Initialize
[STARTED] Install dependencies
[DATA] Installing devDependency @types/node...
[DATA] Installing devDependency @typescript-eslint/eslint-plugin...
[DATA] Installing devDependency @typescript-eslint/parser...
[DATA] Installing devDependency eslint...
[DATA] Installing devDependency eslint-config-prettier...
[DATA] Installing devDependency eslint-plugin-prettier...
[DATA] Installing devDependency prettier...
[DATA] Installing devDependency typescript...
[SUCCESS] Install dependencies
[STARTED] Create tsconfig.json
[SUCCESS] Create tsconfig.json
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
[STARTED] Convert JS files to TS
[DATA] git mv /foo.js to /foo.ts
[DATA] processing file: /foo.ts
[DATA] ? Please view the migration changes for <tmp-path>/foo.js and select an option to continue:
[DATA]  Accept
[DATA]   Edit
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/foo.js and select an option to continue:
[DATA]   Accept
[TITLE] Migration Complete
[TITLE]   1 JS file converted to TS
[TITLE]   1 errors caught by rehearsal
[TITLE]   1 have been fixed by rehearsal
[TITLE]   0 errors need to be fixed manually
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE]     -- 0 eslint errors, with details in the report
[DATA] git mv /depends-on-foo.js to /depends-on-foo.ts
[DATA] processing file: /foo.ts
[DATA] processing file: /depends-on-foo.ts
[DATA] ? Please view the migration changes for <tmp-path>/depends-on-foo.js and select an option to continue:
[DATA]  Accept
[DATA]   Edit
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/depends-on-foo.js and select an option to continue:
[DATA]   Accept
[TITLE] Migration Complete
[TITLE]   2 JS files converted to TS
[TITLE]   2 errors caught by rehearsal
[TITLE]   1 have been fixed by rehearsal
[TITLE]   1 errors need to be fixed manually
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE]     -- 1 eslint errors, with details in the report
[DATA] git mv /index.js to /index.ts
[DATA] processing file: /foo.ts
[DATA] processing file: /depends-on-foo.ts
[DATA] processing file: /index.ts
[DATA] ? Please view the migration changes for <tmp-path>/index.js and select an option to continue:
[DATA]  Accept
[DATA]   Edit
[DATA]   Discard
[DATA]  Please view the migration changes for <tmp-path>/index.js and select an option to continue:
[DATA]   Edit
[DATA]  Please view the migration changes for <tmp-path>/index.js and select an option to continue:
[DATA]   Edit
[TITLE] Migration Complete
[TITLE]   3 JS files converted to TS
[TITLE]   4 errors caught by rehearsal
[TITLE]   3 have been fixed by rehearsal
[TITLE]   1 errors need to be fixed manually
[TITLE]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE]     -- 1 eslint errors, with details in the report
[SUCCESS] Migration Complete
[SUCCESS]   3 JS files converted to TS
[SUCCESS]   4 errors caught by rehearsal
[SUCCESS]   3 have been fixed by rehearsal
[SUCCESS]   1 errors need to be fixed manually
[SUCCESS]     -- 0 ts errors, marked by @ts-expect-error @rehearsal TODO
[SUCCESS]     -- 1 eslint errors, with details in the report"
`;
